src/index.ts

```diff
@@ -1,65 +1,120 @@
 import { DurableObject } from 'cloudflare:workers';
 
-/**
- * Welcome to Cloudflare Workers! This is your first Durable Objects application.
- *
- * - Run `npm run dev` in your terminal to start a development server
- * - Open a browser tab at http://localhost:8787/ to see your Durable Object in action
- * - Run `npm run deploy` to publish your application
- *
- * Bind resources to your worker in `wrangler.toml`. After adding bindings, a type definition for the
- * `Env` object can be regenerated with `npm run cf-typegen`.
- *
- * Learn more at https://developers.cloudflare.com/durable-objects
- */
-
-/** A Durable Object's behavior is defined in an exported Javascript class */
+// Use npm run cf-typegen to generate the type definitions for the Durable Object
+
+// Worker
+export default {
+	async fetch(request: Request, env: Env, ctx: ExecutionContext): Promise<Response> {
+		if (request.url.endsWith('/websocket')) {
+			// Expect to receive a WebSocket Upgrade request.
+			// If there is one, accept the request and return a WebSocket Response.
+			const upgradeHeader = request.headers.get('Upgrade');
+			if (!upgradeHeader || upgradeHeader !== 'websocket') {
+				return new Response('Durable Object expected Upgrade: websocket', {
+					status: 426,
+				});
+			}
+
+			// This example will refer to the same Durable Object,
+			// since the name "foo" is hardcoded.
+			let id = env.WEBSOCKET_HIBERNATION_SERVER.idFromName('foo');
+			let stub = env.WEBSOCKET_HIBERNATION_SERVER.get(id);
+
+			return stub.fetch(request);
+		}
+
+		return new Response(null, {
+			status: 400,
+			statusText: 'Bad Request',
+			headers: {
+				'Content-Type': 'text/plain',
+			},
+		});
+	},
+};
+
+// Durable Object
 export class WebSocketHibernationServer extends DurableObject {
```

Comment:
I would make this name more descriptive of a topology. `WebSocketBroadcast` seems more informative. I think it would be really helpful to also include an example of this using the standard WebSocket libraries.

Reply:
The main idea is to show different aspects of the Hibernation API. Broadcasting message is more on the educational side. I like to think of it like an end-to-end example that folks can copy and paste if they want to have hibernated WebSockets. Based on the application, you can broadcast message, or not.

Yes, I'll be updating the standard WebSocket example once this is merged. I believe it would be simpler in terms of complexity and won't take much time.